engine/src/main/java/com/redhat/cloud/notifications/db/repositories/EndpointRepository.java

```diff
@@ -160,6 +160,27 @@ public boolean incrementEndpointServerErrors(UUID endpointId, int maxServerError
         }
     }
 
+    /**
+     * Gets the endpoint by its UUID.
+     * @param endpointUuid the UUID of the endpoint.
+     * @return the endpoint found.
+     */
+    public Endpoint findByUuid(final UUID endpointUuid) {
+        final String query =
+                "SELECT " +
+                    "e " +
+                "FROM " +
+                    "Endpoint AS e " +
+                "WHERE " +
+                    "id = :uuid";
```

Comment:
Please check the `orgId` as well here.

Reply:
Done! I removed the repository from the engine since it is no longer needed, but added a similar method on the backend's repository.